### PR TITLE
FEATURE: Add setting to enable inline emoji translation

### PR DESCRIPTION
**Pull request by nbianca**
There is a minor issue which prevents the autocomplete from showing up on just ":" (but works with ":s" for example).

### CODE REVIEW

**Comment by SamSaffron** [comment]
we are going to need a couple of tests here :)

File: config/site_settings.yml [hunk 1]
```diff
@@ -744,6 +744,14 @@ posting:
     default: 0
     locale_default:
       fr: 1
+  enable_inline_emoji_translation:
+    client: true
+    default: true
```
Comment (ZogStriP) [inline config/site_settings.yml]: It's the opposite. It should be `false` by default and enabled only for CJK locales.

File: app/assets/javascripts/pretty-text/engines/discourse-markdown/emoji.js.es6 [hunk 2]
```diff
@@ -188,7 +195,7 @@ function applyEmoji(content, state, emojiUnicodeReplacer, enableShortcuts) {
 
   for (i = 0; i < content.length - 1; i++) {
     let offset = 0;
-    let emojiName = getEmojiName(content, i, state);
+    let emojiName = getEmojiName(content, i, state, inlineEmoji);
```
Comment (jjaffeux) [inline app/assets/javascripts/pretty-text/engines/discourse-markdown/emoji.js.es6]:
```suggestion
    const emojiName = getEmojiName(content, i, state, inlineEmoji);
```

File: app/assets/javascripts/discourse/components/d-editor.js.es6 [hunk 3]
```diff
@@ -410,9 +410,11 @@ export default Ember.Component.extend({
       },
 
       onKeyUp(text, cp) {
-        const matches = /(?:^|[^a-z])(:(?!:).?[\w-]*:?(?!:)(?:t\d?)?:?) ?$/gi.exec(
-          text.substring(0, cp)
-        );
+        const regex = self.siteSettings.enable_inline_emoji_translation
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/components/d-editor.js.es6]: Given we are essentially duplicating the regex her with a minor difference, could you try to have two variables with explicit names, and combine them when needed please?

**Pull request merged**